### PR TITLE
Add automatic recovery property to `aws_launch_template` and `aws_instance`

### PR DESCRIPTION
**Pull request by Nnachevvv**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes : https://github.com/hashicorp/terraform-provider-aws/issues/24039

This PR introduces new property `automatic_recovery_behavior` to `aws_launch_template` and `aws_instance`. More details can be found in [Recover your instance](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-recover.html) .

Output from acceptance testing:

```

 make testacc TESTS=TestAccEC2Instance_automaticRecoveryBehavior PKG=ec2   
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2Instance_automaticRecoveryBehavior'  -timeout 180m
=== RUN   TestAccEC2Instance_automaticRecoveryBehavior
=== PAUSE TestAccEC2Instance_automaticRecoveryBehavior
=== CONT  TestAccEC2Instance_automaticRecoveryBehavior

--- PASS: TestAccEC2Instance_automaticRecoveryBehavior (149.20s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	153.234s


make testacc TESTS=TestAccEC2LaunchTemplate_data PKG=ec2       ==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2LaunchTemplate_data'  -timeout 180m
=== RUN   TestAccEC2LaunchTemplate_data
=== PAUSE TestAccEC2LaunchTemplate_data
=== CONT  TestAccEC2LaunchTemplate_data
--- PASS: TestAccEC2LaunchTemplate_data (44.79s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	48.727s


make testacc TESTS=TestAccEC2InstanceDataSource_automaticRecoveryBehavior PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2InstanceDataSource_automaticRecoveryBehavior'  -timeout 180m
=== RUN   TestAccEC2InstanceDataSource_automaticRecoveryBehavior
=== PAUSE TestAccEC2InstanceDataSource_automaticRecoveryBehavior
=== CONT  TestAccEC2InstanceDataSource_automaticRecoveryBehavior
--- PASS: TestAccEC2InstanceDataSource_automaticRecoveryBehavior (182.71s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	186.685s
...
```

### CODE REVIEW

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccEC2Instance_basic\|TestAccEC2Instance_autoRecovery\|TestAccEC2LaunchTemplate_data\|TestAccEC2InstanceDataSource_autoRecovery\|TestAccEC2LaunchTemplateDataSource_name\|TestAccEC2LaunchTemplate_basic' PKG=ec2 ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 3  -run=TestAccEC2Instance_basic\|TestAccEC2Instance_autoRecovery\|TestAccEC2LaunchTemplate_data\|TestAccEC2InstanceDataSource_autoRecovery\|TestAccEC2LaunchTemplateDataSource_name\|TestAccEC2LaunchTemplate_basic -timeout 180m
=== RUN   TestAccEC2InstanceDataSource_autoRecovery
=== PAUSE TestAccEC2InstanceDataSource_autoRecovery
=== RUN   TestAccEC2Instance_basic
=== PAUSE TestAccEC2Instance_basic
=== RUN   TestAccEC2Instance_autoRecovery
=== PAUSE TestAccEC2Instance_autoRecovery
=== RUN   TestAccEC2LaunchTemplateDataSource_name
=== PAUSE TestAccEC2LaunchTemplateDataSource_name
=== RUN   TestAccEC2LaunchTemplate_basic
=== PAUSE TestAccEC2LaunchTemplate_basic
=== RUN   TestAccEC2LaunchTemplate_data
=== PAUSE TestAccEC2LaunchTemplate_data
=== CONT  TestAccEC2InstanceDataSource_autoRecovery
=== CONT  TestAccEC2LaunchTemplateDataSource_name
=== CONT  TestAccEC2Instance_autoRecovery
--- PASS: TestAccEC2LaunchTemplateDataSource_name (43.68s)
=== CONT  TestAccEC2LaunchTemplate_data
--- PASS: TestAccEC2LaunchTemplate_data (14.99s)
=== CONT  TestAccEC2Instance_basic
--- PASS: TestAccEC2Instance_basic (86.84s)
=== CONT  TestAccEC2LaunchTemplate_basic
--- PASS: TestAccEC2Instance_autoRecovery (155.51s)
--- PASS: TestAccEC2LaunchTemplate_basic (13.69s)
--- PASS: TestAccEC2InstanceDataSource_autoRecovery (178.38s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	182.308s
```

**Comment by ewbankkit** [comment]
@Nnachevvv Thanks for the contribution :tada: :clap:.
To more closely match the underlying AWS API and to future-proof ourselves for any future enhancements I made the new argument part of a new `maintenance_options` configuration block.

**Comment by JohnnyBoi888** [comment]
Appreciate involved for addressing this swiftly! 👏 🙇

File: website/docs/r/instance.html.markdown [hunk 1]
```diff
@@ -231,6 +233,12 @@ The `enclave_options` block supports the following:
 
 For more information, see the documentation on [Nitro Enclaves](https://docs.aws.amazon.com/enclaves/latest/user/nitro-enclave.html).
 
+### Maintenance Options
+
+The `maintenance_options` block supports the following:
+
+* `auto_recovery` - (Optional) The automatic recovery behavior of the Instance. Can be `"default"` or `"dedicated"`. See [Recover your instance](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-recover.html) for more details.
```
Comment (sergei-ivanov) [inline website/docs/r/instance.html.markdown:240]: I believe the two valid values are `default` and `disabled`. `dedicated` is a type of instance tenancy.

Comment (JohnnyBoi888) [reply website/docs/r/instance.html.markdown:240]: @ewbankkit pls see ^^, I think documentation probably needs a quick touch up? :)

Comment (ewbankkit) [reply website/docs/r/instance.html.markdown:240]: Relates https://github.com/hashicorp/terraform-provider-aws/pull/24462.

**Pull request merged**